Source/cielim/Private/RenderingFunctionsLibrary.cpp

```diff
@@ -1,28 +1,58 @@
-
 #include "RenderingFunctionsLibrary.h"
 
 #include <OpenCV/PreOpenCVHeaders.h>
 #include <opencv2/core.hpp>
+#include "opencv2/imgproc.hpp"
 #include <OpenCV/opencv-4.5.5/modules/imgcodecs/include/opencv2/imgcodecs.hpp>
 #include <OpenCV/PostOpenCVHeaders.h>
 
+#include <filesystem>
+#include "CielimLoggingMacros.h"
+
+
 URenderingFunctionsLibrary::URenderingFunctionsLibrary(const FObjectInitializer& ObjectInitializer)
 {
-
+	
 }
 
-void URenderingFunctionsLibrary::ApplyPSF_Gaussian()
-
+FString URenderingFunctionsLibrary::ApplyPSF_Gaussian(FString Filepath, int32 KernelHeight, int32 KernelWidth, double SigmaX, double SigmaY)
 {
+	//NOTE: both dimensions of KernelSize must be odd
+	//Read Image
+
+	FString Filepath_Absolute = IFileManager::Get().ConvertToAbsolutePathForExternalAppForRead(*Filepath);
+	
+	std::string Filepath_Absolute_String = TCHAR_TO_UTF8(*Filepath_Absolute);
+	cv::Mat Image = cv::imread(Filepath_Absolute_String);
+
+	//Init ResultImage
+	cv::Mat ResultImage;
+
+	//Get kernel size information into a nice format
+	cv::Size KernelSize = cv::Size(KernelHeight, KernelWidth);
+
+	//Apply blur
+	cv::GaussianBlur(Image, ResultImage, KernelSize, SigmaX, SigmaY);
+
+	//Save image
+	FString ResultFilepath = FPaths::ProjectDir();
+	ResultFilepath.Append("Result_Images/");
+	ResultFilepath.Append("PSF.jpg");
 	
+	std::string ResultFilepath_String = TCHAR_TO_UTF8(*ResultFilepath);
+	cv::imwrite(ResultFilepath_String, ResultImage);
+	
+	return ResultFilepath;
 }
 
 void URenderingFunctionsLibrary::ApplyCosmicRays()
 {
-	
+	//Do Not Call
+	throw std::logic_error("Function not yet implemented");
 }
 
 void URenderingFunctionsLibrary::ApplyReadNoise()
 {
-	
-}
+	//Do Not Call
+	throw std::logic_error("Function not yet implemented");
+}
```

Comment:
:small_red_triangle_down: Files should have a single trailing new line at the end. Github indicates this by showing the red circle and line to the left.

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
Line added!